test/functionality/dataflow/processing-of-elements/functions/source-tests.ts

```diff
@@ -0,0 +1,250 @@
+		.addEdge('3', '2', EdgeType.Argument, 'always')
+		.addEdge('14', '13', EdgeType.Argument, 'always')
+		.addEdge('10', '9', EdgeType.Argument, 'always')
+		.addEdge('3', BuiltIn, EdgeType.Reads, 'always')
+		.addEdge('10', BuiltIn, EdgeType.Reads, 'always')
+		.addEdge('14', BuiltIn, EdgeType.Reads, 'always')
+		.addEdge('13', '12', EdgeType.Reads, 'always')
+		.addEdge('12', 'simple-3:1-3:6-0', EdgeType.Reads, 'always')
+		.addEdge('simple-3:1-3:6-0', '4', EdgeType.SameDefDef, 'always')
+		.addEdge('4', 'simple-1:1-1:6-0', EdgeType.SameDefDef, 'always')
+	)
+
+	const envWithConditionalN = define(
+		{nodeId: 'simple-1:10-1:15-0', scope: 'local', name: 'N', used: 'always', kind: 'variable', definedAt: 'simple-1:10-1:15-2' },
+		LocalScope,
+		initializeCleanEnvironments()
+	)
+	assertDataflow('conditional', shell, 'if (x) { source("simple") }\ncat(N)', new DataflowGraph()
+		.addVertex({ tag: 'variable-definition', id: 'simple-1:10-1:15-0', name: 'N', scope: LocalScope })
+		.addVertex({
+			tag:         'function-call',
+			name:        'source',
+			id:          '4',
+			environment: initializeCleanEnvironments(),
+			args:        [{
+				nodeId: '3', name: `${UnnamedArgumentPrefix}3`, scope: LocalScope, used: 'always' }
+			],
+			when: 'maybe'
+		})
+		.addVertex({
+			tag:         'function-call',
+			name:        'cat',
+			id:          '10',
+			environment: envWithConditionalN,
+			args:        [{
+				nodeId: '9', name: `${UnnamedArgumentPrefix}9`, scope: LocalScope, used: 'always'
+			}]
+		})
+		.addVertex({tag: 'use', id: '0', name: 'x', scope: LocalScope})
+		.addVertex({tag: 'use', id: '8', name: 'N', environment: envWithConditionalN})
+		.addVertex({tag: 'use', id: '3', name: `${UnnamedArgumentPrefix}3`})
+		.addVertex({tag: 'use', id: '9', name: `${UnnamedArgumentPrefix}9`, environment: envWithConditionalN})
+		.addEdge('4', '3', EdgeType.Argument, 'always')
+		.addEdge('4', BuiltIn, EdgeType.Reads, 'maybe')
+		.addEdge('8', 'simple-1:10-1:15-0', EdgeType.Reads, 'always')
+		.addEdge('9', '8', EdgeType.Reads, 'always')
+		.addEdge('10', '9', EdgeType.Argument, 'always')
+		.addEdge('10', BuiltIn, EdgeType.Reads, 'always')
+	)
+
+	// missing sources should just be ignored
+	assertDataflow('missing source', shell, 'source("missing")', new DataflowGraph()
+		.addVertex({
+			tag:         'function-call',
+			name:        'source',
+			id:          '3',
+			environment: initializeCleanEnvironments(),
+			args:        [{
+				nodeId: '2', name: `${UnnamedArgumentPrefix}2`, scope: LocalScope, used: 'always'
+			}]
+		})
+		.addVertex({tag: 'use', id: '2', name: `${UnnamedArgumentPrefix}2`})
+		.addEdge('3', '2', EdgeType.Argument, 'always')
+		.addEdge('3', BuiltIn, EdgeType.Reads, 'always')
+	)
+
+	const envWithX = define(
+		{nodeId: '0', scope: 'local', name: 'x', used: 'always', kind: 'variable', definedAt: '2' },
+		LocalScope,
+		initializeCleanEnvironments()
+	)
+	const recursive2Prefix = 'recursive2-2:1-2:6-'
```

Comment:
better to construct this using the original function or? I mean now we can start a debate regarding what we want to test but i think the tests are fragile enough as-is.